### DISCUSSION TITLE
Status always 200 on redirect.

### DISCUSSION THREAD

**Discussion by adrian-kriegel** [Xunnamius/next-test-api-route-handler#764] [Q&A] [answered]
<details open><summary><strong>Problem</strong></summary>

I think the title is self-explanatory. Status appears to be 200 in any case. 

</details>

<details><summary><strong>Reproduction steps</strong></summary>
Jest example: 

````typescript

import { testApiHandler } from 'next-test-api-route-handler';

test('redirect', async () => 
{
  await testApiHandler(
    {
      handler: (req, res) => res.redirect(307, 'https://example.com'),
      test: async ({ fetch }) => 
      {
        const res = await fetch();
        // this fails, status is 200
        expect(res.status).toBe(307);
      },
    },
  );
});
````
</details>

**Comment by Xunnamius** [comment, marked as answer]
@adrian-kriegel [The fetch spec](https://fetch.spec.whatwg.org/#concept-request-redirect-mode) dictates that redirects [should reuse the original method and body](https://developer.mozilla.org/en-US/docs/Web/HTTP/Status/307) to query the new resource and [return the response as-is](https://fetch.spec.whatwg.org/#atomic-http-redirect-handling). For example, the following (adapted from your reproduction steps) will return status 404:

```javascript
import { testApiHandler } from 'next-test-api-route-handler';

test('redirect', async () =>
{
  await testApiHandler(
    {
      handler: (req, res) => res.redirect(307, 'https://google.com/404'),
      test: async ({ fetch }) =>
      {
        const res = await fetch();
        // this succeeds
        expect(res.status).toBe(404);
      },
    },
  );
});
```

If you visit https://google.com/404 in your browser, you'll also see that you'll get a 404.

If you want to check that a request was redirected, use the `redirected` property on the response object:

```javascript
test('redirect', async () =>
{
  await testApiHandler(
    {
      handler: (req, res) => res.redirect(307, 'https://example.com'),
      test: async ({ fetch }) =>
      {
        const res = await fetch();
        // this succeeds
        expect(res.redirected).toBeTruthy();
      },
    },
  );
});
```

If you want node-fetch (or any relatively spec-compliant fetch library) not to follow redirects, set the [redirect mode](https://github.com/node-fetch/node-fetch#manual-redirect) to `"manual"` like so:

```javascript
import { testApiHandler } from 'next-test-api-route-handler';

test('redirect', async () =>
{
  await testApiHandler(
    {
      handler: (req, res) => res.redirect(307, 'https://example.com'),
      test: async ({ fetch }) =>
      {
        const res = await fetch({ redirect: 'manual' });
        // this succeeds
        expect(res.status).toBe(307);
      },
    },
  );
});
```
